project/SparkBuild.scala

```diff
@@ -765,16 +750,6 @@ object CopyDependencies {
 
 }
 
-object Java8TestSettings {
-  import BuildCommons._
-
-  lazy val settings = Seq(
-    javacJVMVersion := "1.8",
-    // Targeting Java 8 bytecode is only supported in Scala 2.11.4 and higher:
-    scalacJVMVersion := (if (System.getProperty("scala-2.10") == "true") "1.7" else "1.8")
```

Comment:
I agree with the part where you say "that would be fine", not sure I follow the rest.

But yeah you'll end up with mixed 1.7 and 1.8 bytecode as far as I can see. That's fine. It just means it will take a little longer for the class version exception to be thrown in 1.7 in some cases.